### PR TITLE
Issue #9417: Added Example of AST for TokenTypes.DO_WHILE

### PR DESCRIPTION
**Pull request by theexplorist**
Closes: #9417 
DW

```
$ javac Tester.java

$ cat Tester.java
public class Tester {
	public static void test(int a) {
		do {

		} while (a > 0);
	}
}

$ java -jar checkstyle-8.41-all.jar -T Tester.java 
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> Tester [1:13]
`--OBJBLOCK -> OBJBLOCK [1:20]
    |--LCURLY -> { [1:20]
    |--METHOD_DEF -> METHOD_DEF [2:1]
    |   |--MODIFIERS -> MODIFIERS [2:1]
    |   |   |--LITERAL_PUBLIC -> public [2:1]
    |   |   `--LITERAL_STATIC -> static [2:8]
    |   |--TYPE -> TYPE [2:15]
    |   |   `--LITERAL_VOID -> void [2:15]
    |   |--IDENT -> test [2:20]
    |   |--LPAREN -> ( [2:24]
    |   |--PARAMETERS -> PARAMETERS [2:25]
    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [2:25]
    |   |       |--MODIFIERS -> MODIFIERS [2:25]
    |   |       |--TYPE -> TYPE [2:25]
    |   |       |   `--LITERAL_INT -> int [2:25]
    |   |       `--IDENT -> a [2:29]
    |   |--RPAREN -> ) [2:30]
    |   `--SLIST -> { [2:32]
    |       |--LITERAL_DO -> do [3:2]
    |       |   |--SLIST -> { [3:5]
    |       |   |   `--RCURLY -> } [5:2]
    |       |   |--DO_WHILE -> while [5:4]
    |       |   |--LPAREN -> ( [5:10]
    |       |   |--EXPR -> EXPR [5:13]
    |       |   |   `--GT -> > [5:13]
    |       |   |       |--IDENT -> a [5:11]
    |       |   |       `--NUM_INT -> 0 [5:15]
    |       |   |--RPAREN -> ) [5:16]
    |       |   `--SEMI -> ; [5:17]
    |       `--RCURLY -> } [6:1]
    `--RCURLY -> } [7:0]

Printing a line for the code we care
    |       |--LITERAL_DO -> do [3:2]
    |       |   |--SLIST -> { [3:5]
    |       |   |   `--RCURLY -> } [5:2]
    |       |   |--DO_WHILE -> while [5:4]
    |       |   |--LPAREN -> ( [5:10]
    |       |   |--EXPR -> EXPR [5:13]
    |       |   |   `--GT -> > [5:13]
    |       |   |       |--IDENT -> a [5:11]
    |       |   |       `--NUM_INT -> 0 [5:15]
    |       |   |--RPAREN -> ) [5:16]
    |       |   `--SEMI -> ; [5:17]
```
Expected update for javadoc

```
 --LITERAL_DO -> do
    |--SLIST -> {
    |   `--RCURLY -> }
    |--DO_WHILE -> while
    |--LPAREN -> (
    |--EXPR -> EXPR
    |   `--GT -> >
    |       |--IDENT -> a
    |       `--NUM_INT -> 0
    |--RPAREN -> )
    `--SEMI -> ;
```

### CODE REVIEW

**Review by sabaka — APPROVED** [review]
LGTM

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -1785,6 +1785,27 @@
...
      *
+     * <p>For example:</p>
+     * <pre>
+     * do {
+     *
+     * } while (a &gt; 0);
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * |--LITERAL_DO -&gt; do
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1796]: please remove `|` , it does not lead to anywhere

**Review by romani — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (theexplorist) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1796]: Requested changes done, kindly review the changes please @romani

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
Please do not resolve review comments, just reply "done" to each point.

**Pull request merged**